### PR TITLE
Fix todate

### PR DESCRIPTION
**Pull request by daco101**
Fix ToDate to also accept a datetime, fixes Today to be a date.
This PR also fixes an issue with the date to datetime conversion (ToDateTime).

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change

**Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: src/elm/type.coffee [hunk 1]
```diff
@@ -50,7 +50,12 @@ module.exports.ToDate = class ToDate extends Expression
 
   exec: (ctx) ->
     arg = @execArgs(ctx)
-    if arg? and typeof arg != 'undefined' then Date.parse(arg.toString()) else null
+    if (not arg?) or (arg == undefined)
```
Comment (cmoesel) [inline src/elm/type.coffee]: This is fine, but for future reference, I think `not arg?` will also be true for `undefined` -- so you don't need that second `arg == undefined`.

File: test/elm/convert/test.coffee [hunk 2]
```diff
@@ -142,34 +142,49 @@ describe 'FromDateTime', ->
...
   @beforeEach ->
     setup @, data
 
-  it "should convert ToDate('2015-01-01') to DateTime", ->
+  it "should convert @2015-01-01 to DateTime", ->
     dateTime = @dateDateTime.exec(@ctx)
     dateTime.year.should.equal 2015
     dateTime.month.should.equal 1
     dateTime.day.should.equal 1
+    dateTime.timezoneOffset.should.equal @ctx.getTimezoneOffset()
```
Comment (cmoesel) [inline test/elm/convert/test.coffee]: This test should also check that all the time components (hour, minute, second, millisecond) are `0` (as specified in [Section 3.17](http://cql.hl7.org/09-b-cqlreference.html#todatetime) of the spec).

Looking at our current implementation ([here](https://github.com/cqframework/cql-execution/blob/master/src/elm/type.coffee#L63-L64)  and [here](https://github.com/cqframework/cql-execution/blob/master/src/datatypes/datetime.coffee#L503-L504)), I think that test would actually fail.  I know that's not code you touched in this PR, but you touched the test. ;-)

I actually should have caught this one before.

Comment (daco101) [reply test/elm/convert/test.coffee]: Good point. Fixed.

**Comment by daco101** [comment]
I've made the fix in this PR and updated the description.

File: test/elm/convert/test.coffee [hunk 3]
```diff
@@ -182,6 +186,7 @@ describe 'FromDate', ->
     date.year.should.equal 2015
     date.month.should.equal 1
     date.day.should.equal 1
+    should.not.exist(date[field]) for field in [ 'hour', 'minute', 'second', 'millisecond', 'timezoneOffset' ]
```
[the remark this answers is not in the document]
Comment (daco101) [reply test/elm/convert/test.coffee]: I've added the should.not.exist if that's what you mean?

[the remark this answers is not in the document]
Comment (cmoesel) [reply test/elm/convert/test.coffee]: Oh.  My bad.  I was thinking it was toDateTime, but it's toDate -- so you're right.

**Pull request merged**